controllers/webscrapeController.js

```diff
@@ -1,21 +1,106 @@
-const axios = require("axios");
-const cheerio = require("cheerio");
-const { url } = require("./constants");
+const puppeteer = require("puppeteer");
+const { espnUrl, espnScoreUrl } = require("./constants");
 
-async function scrapeEspn(req, res) {
+async function scrapeEspn(startDate, endDate, req, res) {
   try {
-    console.log("Request received for /scrape-espn");
-    const response = await axios.get(url);
-    const $ = cheerio.load(response.data);
+    const browser = await puppeteer.launch();
+    const page = await browser.newPage();
 
-    const headlines = [];
+    await page.setUserAgent('Mozilla/5.0 (iPhone; CPU iPhone OS 7_4_0; like Mac OS X) AppleWebKit/534.49 (KHTML, like Gecko)  Chrome/53.0.1780.199 Mobile Safari/533.3');
 
-    $('.headlineStack__list li').each((index, element) => {
-      const headline = $(element).text().trim();
-      headlines.push(headline);
-    });
+    const allMatchData = [];
 
-    res.json({ headlines });
+    const start = new Date(
+      parseInt(startDate.substring(0, 4)),
+      parseInt(startDate.substring(4, 6)) - 1,
+      parseInt(startDate.substring(6, 8))
+    );
+    
+    const end = new Date(
+      parseInt(endDate.substring(0, 4)),
+      parseInt(endDate.substring(4, 6)) - 1,
+      parseInt(endDate.substring(6, 8))
+    );
+    
+    let current = new Date(start);
+
+    while (current <= end) {
+      const year = current.getFullYear();
+      const month = String(current.getMonth() + 1).padStart(2, '0');
+      const day = String(current.getDate()).padStart(2, '0');
+      const currentDate = `${year}${month}${day}`;
+    
+      const url = `${espnScoreUrl}/_/date/${currentDate}`;
+      
+      await page.goto(url);
+      await page.waitForSelector(".Scoreboard__RowContainer", { timeout: 120000 });
+
+      const matchData = await page.evaluate((date) => {
+        const matchData = [];
+      
+        const matchElements = document.querySelectorAll(".Scoreboard__RowContainer");
+      
+        matchElements.forEach((container) => {
+          const matchInfo = {};
+      
+          const dateString = date.toString();
+          const matchDateObj = new Date(
+            parseInt(dateString.substring(0, 4)),
+            parseInt(dateString.substring(4, 6)) - 1,
+            parseInt(dateString.substring(6, 8))
+          );
+          const options = { year: 'numeric', month: 'long', day: 'numeric' };
+          matchInfo.matchDate = matchDateObj.toLocaleDateString('en-US', options);
+          
+          const leagueElement = container.closest(".Card")?.querySelector(".Card__Header__Title");
+          matchInfo.league = leagueElement ? leagueElement.textContent.trim() : '';
```

Comment:
this can be rewritten as
 matchInfo.league = leagueElement?.textContent?.trim() ?? '';